extensions/resource-deployment/src/services/valueProviderService.ts

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as rd from 'resource-deployment';
+import * as loc from '../localizedConstants';
+
+class ValueProviderService {
+	private _valueProviderStore = new Map<string, rd.IValueProvider>();
+	registerValueProvider(provider: rd.IValueProvider): void {
```

Comment:
This should return a a disposable, and provider should be removed when disposed. for example, when an extension is uninstalled or disabled.

Reply:
Yeah good call, ok let me get that fixed here and in the options source provider and do that in a separate PR